### PR TITLE
added tracking fields for navController

### PR DESCRIPTION
**Pull request by dacianroman**

### CODE REVIEW

File: src/source/view/transitions/BaseTransition.bs [hunk 1]
```diff
@@ -82,6 +82,10 @@ namespace mv.transitions
       if m.source <> invalid and not m.source.isSameNode(m.target)
         m.source@._setShown(false)
         m.owner.removeChild(m.source)
+        if m.owner.poppedView <> invalid
+          m.owner.poppedView = invalid
```
Comment (georgejecook) [inline src/source/view/transitions/BaseTransition.bs:86]: why guard here? I think we can just set it.

File: src/source/view/transitions/BaseTransition.bs [hunk 2]
```diff
@@ -90,6 +94,10 @@ namespace mv.transitions
 
         if m.owner.isShown
           m.target@._setShown(true)
+          if m.owner.pushedView <> invalid
```
Comment (georgejecook) [inline src/source/view/transitions/BaseTransition.bs:97]: why guard?

File: src/source/view/transitions/FadeTransition.bs [hunk 3]
```diff
@@ -40,13 +40,20 @@ namespace mv.transitions
         if m.source <> invalid and not m.source.isSameNode(m.target)
           m.source@._setShown(false)
           m.owner.removeChild(m.source)
+
+          if m.owner.poppedView <> invalid
```
Comment (georgejecook) [inline src/source/view/transitions/FadeTransition.bs:44]: again - why guard?

File: src/source/view/transitions/FadeTransition.bs [hunk 4]
```diff
@@ -40,13 +40,20 @@ namespace mv.transitions
...
         end if
         if m.target <> invalid and not m.target.isSameNode(m.source)
           m.owner.appendChild(m.target)
           m.owner.appendChild(m.fadeRectangle)
-          'TODO - investigate if m.owner.isShown is more accurate
-          ' m.target@._setShown(true)
+
           m.target@._setShown(m.owner.isShown)
+
+          if m.owner.pushedView <> invalid
```
Comment (georgejecook) [inline src/source/view/transitions/FadeTransition.bs:54]: and again.. :)

**Review by georgejecook — APPROVED** [review]
approved; but got questions about the if guards

File: src/source/view/transitions/BaseTransition.bs (hunk 2 above)
Comment (dacianroman) [reply src/source/view/transitions/BaseTransition.bs:97]: true .  i was thinking that other "owners" might not have the fields, but it doesn't matter anyhow. making the changes now

**Review by georgejecook — APPROVED** [review]

**Pull request merged**